src/typechecker/parse-type-definition.lisp

```diff
@@ -192,6 +195,23 @@ Returns TYPE-DEFINITIONS"
                       :constructor-types ctor-types
                       :docstring docstring)))
 
+                  ((and (eql repr :transparent) (not newtype))
+                   (error "Type ~A cannot be repr transparent" tycon-name))
```

Comment:
Is this the error shown to the end user when a type cannot be made transparent? If so it should include some explanation about how the type must be a newtype to be transparent.